gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/DeploymentDirInclusion.java

```diff
@@ -56,6 +56,12 @@ public static void includeFromDeploymentDirs(
                 throw new IllegalStateException("The file set to be the value of getConfigurationYml() "
                         + "must be called configuration.yml. Instead, it was called " + file.getAsFile());
             }));
+
+            // This is a bit of hack to "fall back" to a `build/deployment/configuration.yml` if the
+            // `deployment/configuration.yml` does not exist. This can happen when there's a templating setup
+            // that pre-exists the being able to configuration.yml on the extension and it outputs to
+            // `build/deployment/configuration.yml`.
+            t.from(projectLayout.getBuildDirectory().file("deployment/configuration.yml"));
         });
```

Comment:
It would be really nice if there was a way to evaluate the files copied as part of this copyspec and check that we do actually have a `configuration.yml` in there. I don't think this is possible at the copyspec level (even using `eachFile` - there's no time for a check at the "end" to happen as we never get told when the "end" is).

I think it would need to be a check after the distTar task runs, which even then is dangerous as that check doesn't run before (and so prevent) the publish after it.

Reply:
I guess we could add a task action to all the tar tasks that checks the tar files, but this seems like it might be expensive to read all the way through the tar.gz and I'm concerned it might turn of parallelisation for those tasks.

Reply:
> It would be really nice if there was a way to evaluate the files copied as part of this copyspec and check that we do actually have a `configuration.yml` in there. I don't think this is possible at the copyspec level (even using `eachFile` - there's no time for a check at the "end" to happen as we never get told when the "end" is).
> 
> I think it would need to be a check after the distTar task runs, which even then is dangerous as that check doesn't run before (and so prevent) the publish after it.

I was initially thinking of doing a finaliser task, but it's kinda weird because you might not want to run all the tasks that it might depend on.

You could potentially do something like:
* you have a checking task, it is configured as `mustRunAfter` for all the `distTar` tasks.
* you *always* run the checking task
* it's input should be the distTar output with `getLocationOnly` if possible or we can just annotate the `ConfigurableFileCollection` or whatever type with `@Internal` 
* Can potentially make it incremental

I believe these should allay your concerns? It doesn't feel great though, just a lot of weird hand holding. Might be worth just trying with the `doLast` task action and seeing whether we can get things to work in parallel that way.

Reply:
okay, incremental might not work due to needing to declare it as an input.

Maybe instead of one checking finaliser task, you could just have a finaliser task per `distTar` perhaps, idk

Reply:
you could also potentially have the `eachFile` write into some structure that's an internal property on the finaliser-type task

Reply:
I think finalizers don't work as the publish tasks after will still run if the finalizers fail (under the presence of `--continue`?)

Reply:
I'm going to try and see if adding a `doLast` breaks parallelism

Reply:
that's the behaviour regardless of `--continue`, I think you have to pass in the task state of the `distTar` and check that in the `onlyIf`